README.md

```diff
@@ -15,8 +15,80 @@ following entitlements are supported:
+Ubuntu Advantage client, hereafter "UA client", is python3-based command line utility. It provides a CLI to attach, detach, enable,
+disable and check status of support related services.
+
+The package `ubuntu-advantage-tools` also provides a C++ APT hook which helps advertise ESM service and available packages in MOTD and during various apt commands.
+
+The `ubuntu-advantage-pro` package delivers auto-attach auto-enable functionality via init scripts and systemd services for various cloud platforms.
+
+By default, Ubuntu machines are deployed in an unattached state. A machine can get manually or automatically attached to a specific contract by interacting with the Contract Server REST API to enable a machine. Any change in state of services or machinbe attach results in additional interactions with the Contract Server API to validate such operations. The contract server API is described by the [ua-contracts openapi spec](https://github.com/CanonicalLtd/ua-contracts/blob/develop/docs/contracts.yaml).
+
+### Attaching a machine
+Each Ubuntu SSO account holder has access to one one or more contracts. To attach a machine to an Ubuntu Advantage contract:
+
+* An Ubuntu SSO account holder must obtain a contract token from https://ubuntu.com/advantage.
+* Run `ua attach <contractToken>` on the machine or provide comparable
+* UA Client reads config from /etc/ubuntu-advantage/uaclient.conf nad  a request containing the `<contractToken>` to the Contract Server API @ https://<contract_url>/api/v1/context/machines/token and obtains an unique machine-token JSON response containing credentials, affordances, directives and obligations to allow enabling and disabling any Ubuntu Advantage service
+* UA Client POSTs a request containing the `<contractToken>` to the Contract Server API @ https://contracts.canonical.com/api/v1/context/machines/tokento to obtain an unique machine-token JSON response which contains credentials, affordances, directives and obligations to allow enabling and disabling any Ubuntu Advantage service
+* UA client auto-enables any services described with `obligations:{enableByDefault: true}`
+POST `<contractToken>` to contracts server API @ https://contracts.canonical.com/api/v1/context/machines/token
+* UA client writes the API response to the root-readonly /var/lib/ubuntu-advantage/machine-token.json
+
+### Enabling a service
+Each service controlled by UA client will have a python module in uaclient/entitlements/\*.py which handles setup and teardown of services when enabled or disabled.
+
+If a contract entitles a machine to a service, `root` user can enable the service with `ua enable <service>`.  If a service can be disabled `ua disabled <service>` will be permitted.
+
+The goal of the UA client is to remain simple and flexible and let the
+contracts backend drive dynamic changes in contract offerings and contraints.
+In pursuit of that goal, the UA client obtains most of it's service contraints
+from a machine token that it ob
```

Comment:
I think this paragraph is incomplete

Reply:
Fixed ... "from a machine token that it obtains from the Contract Server API. "